clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/VenicePushJobTest.java

```diff
@@ -150,6 +147,31 @@ public void testPushJobSettingWithLivenessHeartbeat() {
     Assert.assertTrue(pushJobSetting.livenessHeartbeatEnabled);
   }
 
+  @Test
+  public void testPushJobPollStatus() {
+    Properties vpjProps = new Properties();
+    vpjProps.setProperty(HEARTBEAT_ENABLED_CONFIG.getConfigName(), "true");
+    ControllerClient client = mock(ControllerClient.class);
+    JobStatusQueryResponse response = mock(JobStatusQueryResponse.class);
+    doReturn("UNKNOWN").when(response).getStatus();
+    doReturn(response).when(client).queryOverallJobStatus(anyString(), eq(Optional.empty()), eq(null));
+    VenicePushJob pushJob = getSpyVenicePushJob(vpjProps, client);
+    VenicePushJob.PushJobSetting pushJobSetting = pushJob.getPushJobSetting();
+    pushJobSetting.jobStatusInUnknownStateTimeoutMs = 10;
+    Assert.assertTrue(pushJobSetting.livenessHeartbeatEnabled);
+    VenicePushJob.TopicInfo topicInfo = new VenicePushJob.TopicInfo();
+    topicInfo.version = 1;
+    topicInfo.topic = "abc";
+    pushJob.storeSetting = new VenicePushJob.StoreSetting();
+    pushJob.storeSetting.storeResponse = new StoreResponse();
+    StoreInfo storeInfo = new StoreInfo();
+    storeInfo.setBootstrapToOnlineTimeoutInHours(1);
+    pushJob.storeSetting.storeResponse.setStore(storeInfo);
+    Assert.expectThrows(
+        VeniceException.class,
+        () -> pushJob.pollStatusUntilComplete(Optional.empty(), client, pushJobSetting, topicInfo));
```

Comment:
Can we also assert the exception thrown contain the right msg? It seems like the exception is thrown due to being in UNKNOWN state instead of being over the time limit set by `bootstrapToOnlineTimeoutInHours`.

In terms of testing, maybe we can set `BootstrapToOnlineTimeoutInHours` to be 0 and compare the time difference in milliseconds instead of hour so it's easier to reproduce this in tests.

Reply:
From this [javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/TimeUnit.html#convert-long-java.util.concurrent.TimeUnit-)
>  Conversions from finer to coarser granularities truncate, so lose precision. For example, converting 999 milliseconds to seconds results in 0

So, I think it would always be 0 unless it hits one hour mark, I haven't tested it though.

Reply:
yeah it will be 0 if elapsed time is < 1hr.  we can either set the config to -1 or convert everything to millisec and compare to trigger the code.

Reply:
Yeah. Personally converting hours->ms might be better I guess...